lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -174,9 +190,23 @@ def get_results(self, name_list: List[str], query: JSON, server_config) -> JSON:
                 ["test", "fio_1", "uperf_1", "uperf_2", "uperf_3", "uperf_4"],
             ),
             ("drb", {"end": "1970-09-01"}, []),
+            ("drb", {"filter": "dataset.access:public"}, ["fio_1", "fio_2"]),
+            ("drb", {"filter": "dataset.name:fio_1"}, ["fio_1"]),
+            (
+                "test",
+                {"filter": "^dataset.name:~fio,^dataset.name:uperf_1"},
+                ["fio_1", "fio_2", "uperf_1"],
+            ),
+            (
+                "test",
+                {
+                    "filter": "^dataset.name:~fio,^dataset.name:uperf_1,dataset.owner_id:3"
+                },
+                ["fio_1"],
+            ),
```

Comment:
Do the above additions cover all paths through the new filter code?

Reply:
Mr. Jenkins says [not quite](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/job/pbench-ci-multi/job/PR-3213/8/cobertura/server_api_resources/datasets_list_py/):  aside from some user error conditions, it looks like we're not covering `native_key == Metadata.USER` nor any other non-`Metadata.DATASET` keys.

Reply:
... which is what I said... 😁

Reply:
Yeah, but you said it _after_ I said it...at least according to the timestamps.  😉

Reply:
> Yeah, but you said it _after_ I said it...at least according to the timestamps. 😉

Oh, no, the time is running the other way.  😝 

Anyway, I think you and I were composing our replies at the same time...mine just ended up posted later.